src/main/java/org/mule/extension/sftp/internal/SftpInputStream.java

```diff
@@ -46,132 +44,89 @@ protected static ConnectionManager getConnectionManager(SftpConnector config) th
       this.connectionManager = connectionManager;
-      this.timeBetweenSizeCheck = timeBetweenSizeCheck;
       this.config = config;
     }
 
     @Override
-    public InputStream get() {
+    protected FileAttributes getUpdatedAttributes() {
       try {
-        SftpFileAttributes updatedAttributes = getUpdatedAttributes(config, connectionManager, attributes.getPath());
-        if (updatedAttributes != null && timeBetweenSizeCheck != null && timeBetweenSizeCheck > 0) {
-          updatedAttributes = getUpdatedStableAttributes(config, connectionManager, updatedAttributes);
-        }
-        if (updatedAttributes == null) {
-          throw new DeletedFileWhileReadException(createStaticMessage("File on path " + attributes.getPath()
-              + " was read but does not exist anymore."));
+        ConnectionHandler<SftpFileSystem> connectionHandler = connectionManager.getConnection(config);
+        SftpFileSystem sftpFileSystem = connectionHandler.getConnection();
+        SftpFileAttributes updatedSftpFileAttributes = sftpFileSystem.readFileAttributes(attributes.getPath());
+        connectionHandler.release();
+        if (updatedSftpFileAttributes == null) {
+          LOGGER.error(String.format(FILE_NO_LONGER_EXISTS_MESSAGE, attributes.getPath()));
         }
-        connectionHandler = connectionManager.getConnection(config);
-        return connectionHandler.getConnection().retrieveFileContent(updatedAttributes);
+        return updatedSftpFileAttributes;
       } catch (ConnectionException e) {
         throw new MuleRuntimeException(createStaticMessage("Could not obtain connection to fetch file " + attributes.getPath()),
                                        e);
       }
     }
 
-    private SftpFileAttributes getUpdatedStableAttributes(SftpConnector config, ConnectionManager connectionManager,
-                                                          SftpFileAttributes updatedAttributes)
-        throws ConnectionException {
-      SftpFileAttributes oldAttributes;
-      int retries = 0;
-      do {
-        oldAttributes = updatedAttributes;
-        try {
-          if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug(String.format(STARTING_WAIT_MESSAGE, attributes.getPath()));
-          }
-          sleep(timeBetweenSizeCheck);
-        } catch (InterruptedException e) {
-          throw new MuleRuntimeException(createStaticMessage("Execution was interrupted while waiting to recheck file sizes"),
-                                         e);
+    @Override
+    protected InputStream getContentInputStream() {
+      try {
+        connectionHandler = connectionManager.getConnection(config);
+        sftpFileSystem = connectionHandler.getConnection();
+        return sftpFileSystem.retrieveFileContent(attributes);
+      } catch (MuleRuntimeException e) {
+        if (e.getCause() instanceof SftpException && e.getCause().getMessage().contains("FileNotFoundException:")) {
```

Comment:
method `fileWasDeleted`, and change `FileNotFoundException` with `FileNotFoundException.class.getSimpleName()`, removing the trailing `:`